mainwindow.cpp

```diff
@@ -120,28 +133,48 @@ void MainWindow::load_config(QString filename) {
 		}
 
 		// ----------------------------
-		// recording location
+		// Block/Task Names
 		// ----------------------------
-		ui->blockList->addItems(pt.value("SessionBlocks").toStringList());
-		if (ui->blockList->count()) currentBlock = ui->blockList->itemText(0);
+		// Clear out any widgets where block/task name goes.
+		QLayoutItem *child = ui->horizontalLayout_blockTask->takeAt(1);
+		if (child != 0) { delete child; }
+		QStringList taskNames;
+		if (pt.contains("SessionBlocks")) { taskNames = pt.value("SessionBlocks").toStringList(); }
+		hasTasks = taskNames.count() > 0;
+		if (hasTasks) {
+			taskBox = new QComboBox();
```

Comment:
Couldn't we have a settings container with two tabs and then switch between them depending on BIDS/non-BIDS-mode?

Reply:
The presence of the combo box isn't determined by BIDS/other, it's determined by whether or not the cfg file had a SessionBlocks section to define the blocks/tasks. If there was no such block we could leave the comboxbox empty then it becomes impossible to fill in the %b placeholder. However, if we replace the combobox with a LineEdit field then the user can type in the name of the block/task. This is an enhancement that would apply to both BIDS and Legacy.